tests/integration/api/test_activation.py

```diff
@@ -365,6 +374,25 @@ def test_list_activations_filter_decision_environment_id(client: APIClient):
     assert len(response.data["results"]) == 0
 
 
+@pytest.mark.django_db
+def test_list_activations_filter_credential_id(client: APIClient) -> None:
+    """Test filtering by credential_id."""
+    # TODO(alex): Refactor the presetup, it should be fixtures
+    fks = create_activation_related_data()
+    create_activation(fks)
+    credential_id = fks["credential_id"]
+
+    url = f"{api_url_v1}/activations/?credential_id={credential_id}"
+    response = client.get(url)
+    assert response.status_code == status.HTTP_200_OK
+    assert len(response.data["results"]) == 1
+
+    url = f"{api_url_v1}/activations/?credential_id=2"
```

Comment:
This might fail if we have more than 1 credentials for tests in the future. Should we use an arbitrary number like 0 to work for all cases?

Reply:
I don't think so, the test uses a clean DB. I have increased the number just in case. Anyway I want to rework the fixtures thing in later PR's